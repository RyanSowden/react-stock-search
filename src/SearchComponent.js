import React, {useState, useEffect} from 'react';
import ReactDom from 'react-dom';
import axios from 'axios';
import {Line, Bar} from 'react-chartjs-2';
import Chart from 'chart.js/auto';

const SearchComponent = () =>  {

	const [ticker,setTicker] = useState('')
	const [currentStock,setCurrentStock] = useState('')
	const [adjustedStock,setAdjustedStock] = useState('')
	
	/*
	componentDidMount(){
		const [data,setData] = useState({});

		useEffect(() => {
			setData({
				labels: ['Jan','Feb','Mar'],
				datasets: [
					{
				label: 'Test',
					fill: false,
					lineTension: 0.5,
					backgroundColor: 'pink',
					borderColor: 'black',
					borderWidth: 2,
					data: [1,2,3]
			}
				]
		});
},[])
	}
*/
	
	const handleChange = (e) => { 
		setTicker(e.target.value); 
	}

	const getCurrentPrice = async ()  =>  {
		const stock = ticker 
		try {
			const current_response = await axios.post('http://localhost:2000/current',{ticker});
			const current_result = await current_response
			//const history_result = await history_response
			let current_price = (current_result['data'].regularMarketPrice)
			//let adjust_price = (current_result['data'].postMarketPrice)
			//let history_price = (history_result['data'][0].close)
			setCurrentStock(current_price)

		} catch(err) {
			console.log(err)
		};
	}


	const getAjustedPrice = async ()  =>  {
		const stock = ticker 
		try {
			const adjusted_response = await axios.post('http://localhost:2000/current',{ticker});
			const adjusted_result = await adjusted_response
			//const history_result = await history_response
			let adjusted_price = (adjusted_result['data'].postMarketPrice)
			//let history_price = (history_result['data'][0].close)
			setAdjustedStock(adjusted_price)

		} catch(err) {
			console.log(err)
		};
	}




	const handleSubmit = (e) => {
		e.preventDefault();
		getCurrentPrice();
		getAjustedPrice();
	};


	return (
		<div>
			<form onSubmit={handleSubmit} id="form">
			<input 
			type="text"
			value={ticker}
			onChange={handleChange}
			id="search"
			placeholder="Stock Ticker"
			name="search"
			/>
			<button id="searchBtn" type="submit">Search</button>
		</form>
			<p>{currentStock}</p>
			<p>{adjustedStock}</p>

		</div>
	);

}



export default SearchComponent


