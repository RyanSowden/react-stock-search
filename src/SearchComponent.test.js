import React from 'react';
import SearchComponent from './SearchComponent';
import Enzyme from 'enzyme';
import {shallow, mount} from 'enzyme';
import {render, screen, fireEvent} from '@testing-library/react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

describe("Button test", () => {
	it("should initialize index to 0", () => {
    	// check for initial state
    	const app = shallow(<SearchComponent />);
    	expect(app.state("value")).toEqual("");
  });

	//check for the state to be updated.
	it("Check for updates to the state value", () => {
		const app = shallow(<SearchComponent />);
		app.instance().setState({value: 'test'});
		expect(app.state('value')).toEqual('test');
	
	});
	it("Check if the current_price variable takes a value", () => {
		const app = shallow(<SearchComponent />);
		const current_price = 10;
		app.instance().setState({current_price: current_price});
		expect(app.state('current_price')).toEqual(10)
	});

	it("Check if the current_price variable takes a value", () => {
		const app = shallow(<SearchComponent />);
		const adjust_price = 10;
		app.instance().setState({adjust_price: adjust_price});
		expect(app.state('adjust_price')).toEqual(10)
	});

});
