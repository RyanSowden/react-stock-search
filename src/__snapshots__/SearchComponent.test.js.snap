// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent should render properly 1`] = `
<div>
  <form
    id="form"
    onSubmit={[Function]}
  >
    <input
      id="search"
      name="search"
      onChange={[Function]}
      placeholder="Stock Ticker"
      type="text"
      value=""
    />
    <button
      id="searchBtn"
      type="submit"
    >
      Search
    </button>
  </form>
  <p>
    
  </p>
</div>
`;
